### PR TITLE
comparison and xpath breakdown policy updates

### PR DESCRIPTION
**Pull request by dericed**

### CODE REVIEW

**Comment by dericed** [comment]
This is a WIP, do not merge yet. cc @ablwr

**Comment by dericed** [comment]
done with WIP phase, please review

File: Tools/Policies/policyset2policytransform2policyresult.xsl [hunk 1]
```diff
@@ -17,15 +17,15 @@
               </aliasxsl:attribute>
               <xsl:if test="string-length($compare)>0">
                 <aliasxsl:attribute name="compare">
-                  <xsl:value-of select="$compare"/>
+                  <xsl:value-of select="document($compare)//ma:media[1]/@ref"/>
                 </aliasxsl:attribute>
               </xsl:if>
-              <xsl:for-each select="policy">
+              <xsl:for-each select="mp:policy">
```
Comment (ablwr) [inline Tools/Policies/policyset2policytransform2policyresult.xsl]: +1 on getting more granular by using the declared namespace here and below.

**Comment by dericed** [comment]
What is the status of this? +1's from me and @ablwr and then @JeromeMartinez changed the title back to include "WIP"

**Comment by ablwr** [comment]
@dericed also used the hooray emoji 🎉 !

**Comment by JeromeMartinez** [comment]
ha, some synchronization issue, I wanted to change "WIP]" to "[WIP]"!

**Comment by JeromeMartinez** [comment]
@tribouille please review

**Comment by tribouille** [comment]
I try with the first policy example (sample_policy_1.xml)... Not working
Same for the 4th... And did not try other...

resulting XML==
<?xml version="1.0" encoding="UTF-8"?>
<MediaConch xmlns="https://mediaarea.net/mediaconch" xmlns:mt="https://mediaarea.net/mediatrace" xmlns:mmt="https://mediaarea.net/micromediatrace" xmlns:mi="https://mediaarea.net/mediainfo" version="0.3">
  <media ref="/tmp/test.mkv"/>
</MediaConch>

**Comment by dericed** [comment]
Issue is this change should be applied to other policies. https://github.com/dericed/MediaConch_SourceCode/commit/300655675b881dfb1866909155f0168cfa4aa1f0 However I thought I had already put that in another pr. Looking for that now.

**Comment by dericed** [comment]
@tribouille PR updated, please retest

**Comment by dericed** [comment]
@tribouille @JeromeMartinez I dropped some commits, rebase, and force pushed to remove the use of mediapolicy namespace. Please test.

**Comment by tribouille** [comment]
Working with the basic policies.
Problem with the compare policies.

`
./mediaconch -fx -p ../../../Tools/Policies/sample_policy_2_comparison.xml -prf /tmp/test.mkv /tmp/test.mkv 
<?xml version="1.0" encoding="UTF-8"?>
<MediaConch xmlns="https://mediaarea.net/mediaconch" xmlns:mt="https://mediaarea.net/mediatrace" xmlns:mmt="https://mediaarea.net/micromediatrace" xmlns:mi="https://mediaarea.net/mediainfo" version="0.3">
  <media ref="/tmp/test.mkv" compare="/tmp/test.mkv">
    <policy name="Smaller frame size and same aspect ratio?" type="and" rules_run="3" fail_count="2" pass_count="1" outcome="fail">
      <description>Test to see if a file has a smaller frame size than another file but same aspect ratio.</description>
      <rule name="Width is less?" value="Width" tracktype="Video" occurrence="1" operator="&lt;" xpath="mi:MediaInfo/mi:track[@type='Video'][1]/mi:Width&lt;document('/home/tribou_f/.local/share/MediaConch/policies_references_files/reference3.xml')//mi:MediaInfo/mi:track[@type='Video'][1]/mi:Width" actual="640" compared_to="640" outcome="fail"/>
      <rule name="Height is less?" value="Height" tracktype="Video" occurrence="1" operator="&lt;" xpath="mi:MediaInfo/mi:track[@type='Video'][1]/mi:Height&lt;document('/home/tribou_f/.local/share/MediaConch/policies_references_files/reference3.xml')//mi:MediaInfo/mi:track[@type='Video'][1]/mi:Height" actual="480" compared_to="480" outcome="fail"/>
      <rule name="Display Aspect Ratio is same?" value="DisplayAspectRatio" tracktype="Video" occurrence="1" operator="=" xpath="mi:MediaInfo/mi:track[@type='Video'][1]/mi:DisplayAspectRatio=document('/home/tribou_f/.local/share/MediaConch/policies_references_files/reference3.xml')//mi:MediaInfo/mi:track[@type='Video'][1]/mi:DisplayAspectRatio" outcome="pass"/>
    </policy>
  </media>
</MediaConch>
`

Is it in 431 or 432?

**Comment by dericed** [comment]
@tribouille I don't understand, the output looks logical. The width of fileA is not "less than" width of fileB (since fileA=fileB).

**Comment by tribouille** [comment]
Ok for me

**Pull request merged**